contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,419 @@
 
-For the initial implementation we will only support limits for cpu and memory resources.
+* cpu.shares = (cpu in millicores * 1024) / 1000
+* cpu.cfs_period_us = 100000 (i.e. 100ms)
+* cpu.cfs_quota_us = quota = (cpu in millicores * 100000) / 1000
 
-#### QoS classes
+## Pod level cgroups
 
-A pod can belong to one of the following 3 QoS classes: Guaranteed, Burstable, and BestEffort, in decreasing order of priority.
+The `kubelet` will create a cgroup sandbox for each pod.
 
-#### Guaranteed
+The naming convention for the cgroup sandbox is `pod<pod.UID>`.  It enables
+the `kubelet` to associate a particular cgroup on the host filesytem
+with a corresponding pod without managing any additional state.  This is useful
+when the `kubelet` restarts and needs to verify the cgroup filesystem.
 
-`G` pods will be placed at the `$Root` cgroup by default. `$Root` is the system root i.e. "/" by default and if `--cgroup-root` flag is used then we use the specified cgroup-root as the `$Root`. To ensure Kubelet's idempotent behaviour we follow a pod cgroup naming format which is opaque and deterministic. Say we have a pod with UID: `5f9b19c9-3a30-11e6-8eea-28d2444e470d` the pod cgroup PodUID would be named: `pod-5f9b19c93a3011e6-8eea28d2444e470d`.
+A pod can belong to one of the following 3 QoS classes in decreasing priority:
+
+1. Guaranteed
+1. Burstable
+1. BestEffort
+
+The resource configuration for the cgroup sandbox is dependent upon the
+pod's associated QoS class.
+
+### Guaranteed QoS
+
+A pod in this QoS class has its cgroup sandbox configured as follows:
+
+```
+pod<UID>/cpu.shares = sum(pod.spec.containers.resources.requests[cpu])
+pod<UID>/cpu.cfs_quota_us = sum(pod.spec.containers.resources.limits[cpu])
+pod<UID>/memory.limit_in_bytes = sum(pod.spec.containers.resources.limits[memory])
+```
 
+### Burstable QoS
 
-__Note__: The cgroup-root flag would allow the user to configure the root of the QoS cgroup hierarchy. Hence cgroup-root would be redefined as the root of QoS cgroup hierarchy and not containers.
+A pod in this QoS class has its cgroup sandbox configured as follows:
 
 ```
-/PodUID/cpu.quota = cpu limit of Pod  
-/PodUID/cpu.shares = cpu request of Pod  
-/PodUID/memory.limit_in_bytes = memory limit of Pod
+pod<UID>/cpu.shares = sum(pod.spec.containers.resources.requests[cpu])
+```
+
+If all containers in the pod specify a cpu limit:
+
+```
+pod<UID>/cpu.cfs_quota_us = sum(pod.spec.containers.resources.limits[cpu])
+```
+
+Finally, if all containers in the pod specify a memory limit:
+
+```
+pod<UID>/memory.limit_in_bytes = sum(pod.spec.containers.resources.limits[memory])
+```
+
+Note: This design enables containers in a pod to optionally share slack compute resources.
+
+### BestEffort QoS
+
+A pod in this QoS class has its cgroup sandbox configured as follows:
+
 ```
+pod<UID>/cpu.shares = 2
+```
+
+## QoS level cgroups
+
+The `kubelet` defines a `--cgroup-root` flag that is used to specify the `ROOT`
```

Comment:
I'd mention that the recommended configuration is to set `--cgroup-root` to `/` to avoid having a deep cgroup hierarchy.